File: quan.cs/Test/Message/Common/MessageHeader.cs
using Quan.Common.Utils;
using Quan.Message;
using Buffer = Quan.Message.Buffer;

namespace Test.Message.Common
{
	/// <summary>
	/// 消息头<br/>
	/// 代码自动生成，请勿手动修改
	/// </summary>
    public abstract class MessageHeader : MessageBase
    {
        /// <summary>
		/// 消息序号
		/// </summary>
		public long seq { get; set; }

        /// <summary>
		/// 错误码
		/// </summary>
		public int error { get; set; }


		public override void Encode(Buffer buffer)
		{
	    	base.Encode(buffer);

		    buffer.WriteLong(seq);
		    buffer.WriteInt(error);
		}

		public override void Decode(Buffer buffer)
		{
	    	base.Decode(buffer);

		    seq = buffer.ReadLong();
		    error = buffer.ReadInt();
		}

		public override string ToString()
		{
			return "MessageHeader{" +
				   "seq=" + seq.ToString2() +
				   ",error=" + error.ToString2() +
				   '}';
		}
    }
}

File: quan.lua/src/test/message/common/MessageHeader.lua
---
---消息头
---代码自动生成，请勿手动修改
---

local _Buffer = require("quan.message.Buffer")
local _Message = require("quan.message.Message")

---
---消息头
---
local MessageHeader = {
    ---类名
    class = "test.message.common.MessageHeader",
}

---
---[消息头].编码
---@param buffer quan.message.Buffer 可以为空
---@return quan.message.Buffer
---
function MessageHeader:encode(buffer)
    assert(buffer == nil or type(buffer) == "table" and buffer.class == _Buffer.class, "参数[buffer]类型错误")

    buffer = buffer or _Buffer.new()

    buffer:writeLong(self.seq)
    buffer:writeInt(self.error)

    return buffer
end

---
---[消息头].解码
---@param buffer quan.message.Buffer 不能为空
---@param self test.message.common.MessageHeader 可以为空
---@return test.message.common.MessageHeader
---
function MessageHeader.decode(buffer, self)
    assert(type(buffer) == "table" and buffer.class == _Buffer.class, "参数[buffer]类型错误")

    self = self or MessageHeader.new()

    self.seq = buffer:readLong()
    self.error = buffer:readInt()

    return self
end

MessageHeader = table.readOnly(MessageHeader)
return MessageHeader